urld0f681c558_farom57_AstroidDriver_tgz-pJ8-laazotea_indi_server_examples_MinimumINDIServerJ8/src/laazotea/indi/driver/INDITextElement.java

```diff
@@ -0,0 +1,117 @@
+/*
+ *  This file is part of INDI for Java Driver.
+ * 
+ *  INDI for Java Driver is free software: you can redistribute it
+ *  and/or modify it under the terms of the GNU General Public License 
+ *  as published by the Free Software Foundation, either version 3 of 
+ *  the License, or (at your option) any later version.
+ * 
+ *  INDI for Java Driver is distributed in the hope that it will be
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+ *  of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ * 
+ *  You should have received a copy of the GNU General Public License
+ *  along with INDI for Java Driver.  If not, see 
+ *  <http://www.gnu.org/licenses/>.
+ */
+package laazotea.indi.driver;
+
+import org.w3c.dom.Element;
+
+/**
+ * A class representing a INDI Text Element.
+ *
+ * @author S. Alonso (Zerjillo) [zerjioi at ugr.es]
+ * @version 1.36, November 17, 2013
+ */
+public class INDITextElement extends INDIElement {
+
+  /**
+   * The current value of the Text Element
+   */
+  private String value;
+
+  /**
+   * Constructs an instance of a
+   * <code>INDITextElement</code> with a
+   * <code>name</code>, a
+   * <code>label</code> and its initial
+   * <code>value</code>.
+   *
+   * @param property The Property to which this Element belongs.
+   * @param name The name of the Element.
+   * @param label The label of the Element.
+   * @param value The initial value of the Element
+   */
+  public INDITextElement(INDITextProperty property, String name, String label, String value) {
+    super(property, name, label);
+
+    this.value = value.trim();
+  }
+
+  /**
+   * Constructs an instance of a
+   * <code>INDITextElement</code> with a
+   * <code>name</code> and its initial
+   * <code>value</code>. The label of the Element will be a copy of the
+   * <code>name</code>.
+   *
+   * @param property The Property to which this Element belongs.
+   * @param name The name of the Element.
+   * @param value The initial value of the Element
+   * @throws IllegalArgumentException
+   */
+  public INDITextElement(INDITextProperty property, String name, String value) throws IllegalArgumentException {
+    super(property, name);
+
+    this.value = value.trim();
+  }
+
+  @Override
+  public INDITextProperty getProperty() {
+    return (INDITextProperty)super.getProperty();
+  }
+
+  @Override
+  public String getValue() {
+    return value;
+  }
+
+  @Override
+  public void setValue(Object newValue) throws IllegalArgumentException {
+    String v = null;
+
+    try {
+      v = (String)newValue;
+    } catch (ClassCastException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)